### ISSUE TITLE
The -f,--file option does not work with "kv restore"  

### ISSUE THREAD

**Issue by CrankyDragon** [gmr/consulate#33]
When using the -f,--file option with kv restore, the kv_restore function expects that a string, but a file object is being provided:

```
root@vagrant-ubuntu-trusty-64:~# consulate kv restore --file dump.json
Traceback (most recent call last):
  File "/usr/local/bin/consulate", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/consulate/cli.py", line 274, in main
    kv_restore(consul, args)
  File "/usr/local/lib/python2.7/dist-packages/consulate/cli.py", line 196, in kv_restore
    handle = open(args.file, 'r') if args.file else sys.stdin
TypeError: coercing to Unicode: need string or buffer, file found
```

Looks like when --file is specified, the file is opened automatically:
```python
    restore = kvsparsers.add_parser('restore',
                                    help='Restore from stdin or a JSON file')
    restore.add_argument('-f', '--file',
                         help='JSON file to read instead of stdin',
                         nargs="?", type=open)
```

So, simply swapping 
```python
    handle = open(args.file, 'r') if args.file else sys.stdin
```
with:
```python
    handle = args.file
```
in cli.py on line 196 should fix the issue. Piping data in kv restore does work correctly however:
```
cat SOME_FILE.json | consulate kv restore
```

**Comment by gmr** [comment]
Thanks for the analysis. I think this is fixed with the refactor of the argparse setup that was committed tonight. I'll make sure to test this prior to the soon to be released 0.5.0.

**Comment by gmr** [comment]
0.5.0 is released and this should be good to go. Pls reopen if you have any additional issues.
